### PR TITLE
Add springboot example demo

### PR DESCRIPTION
**Pull request by Liujingfang1**
first revision of spring boot example demo

### CODE REVIEW

**Comment by Liujingfang1** [comment]
Updated the demo with configmaps. PTAL.

File: cmd/kinflate/demo/springboot.md [hunk 1]
```diff
@@ -0,0 +1,328 @@
+# Demo: SpringBoot
+
+This example takes k8s resources for one Spring Boot application sbdemo, and customizes them to suit a
+production scenario.
```
Comment (droot) [inline cmd/kinflate/demo/springboot.md]: May be can re-word this:
In this tutorial, you will learn - how to use `kinflate` to customize a basic Sprint Boot application's k8s configuration for dev and production use cases.

File: cmd/kinflate/demo/springboot.md [hunk 2]
```diff
@@ -0,0 +1,328 @@
...
+
+In the production environment we want:
```
Comment (droot) [inline cmd/kinflate/demo/springboot.md]: may be say:
we want to customize the following:

File: cmd/kinflate/demo/springboot.md [hunk 3]
```diff
@@ -0,0 +1,328 @@
...
+
+- add configuration for this Spring Boot application
+- customize configuration by adding prod database connection
```
Comment (droot) [inline cmd/kinflate/demo/springboot.md]: configure prod DB access configuration

File: cmd/kinflate/demo/springboot.md [hunk 4]
```diff
@@ -0,0 +1,328 @@
+# Demo: SpringBoot
+
+This example takes k8s resources for one Spring Boot application sbdemo, and customizes them to suit a
+production scenario.
+
+In the production environment we want:
+
+- add configuration for this Spring Boot application
```
Comment (droot) [inline cmd/kinflate/demo/springboot.md]: Is it "application specific" configuration ?

File: cmd/kinflate/demo/springboot.md [hunk 5]
```diff
@@ -0,0 +1,328 @@
...
+- customize configuration by adding prod database connection
+- resource names to be prefixed by 'prod-'.
+- resources to have 'env: prod' labels.
+- jvm memory to be properly set.
```
Comment (droot) [inline cmd/kinflate/demo/springboot.md]: nit: jvm --> JVM

File: cmd/kinflate/demo/springboot.md [hunk 6]
```diff
@@ -0,0 +1,328 @@
...
+
+Create one:
+
+<!-- @initApp @test -->
+```
+cd $DEMO_HOME
+kinflate init
+```
+
+You should now have a file called `Kube-manifest.yaml`:
```
Comment (droot) [inline cmd/kinflate/demo/springboot.md]: The above step will create a `kinflate` configuration file called `Kube-manifest.yaml` in current directory.

File: cmd/kinflate/demo/springboot.md [hunk 7]
```diff
@@ -0,0 +1,328 @@
...
+
+containing something like:
+
+
+> ```
+> apiVersion: manifest.k8s.io/v1alpha1
+> kind: Manifest
+> metadata:
+>   name: helloworld
+> # description: helloworld does useful stuff.
```
Comment (droot) [inline cmd/kinflate/demo/springboot.md]: I am wondering if we should take out these "hello-world" stuff from the initial Kinflate configuration file ?

File: cmd/kinflate/demo/springboot.md [hunk 8]
```diff
@@ -0,0 +1,328 @@
...
+When a Spring Boot application is deployed in a k8s cluster, the JVM is running inside a container. We want to set memory limit for the container and make sure
+the JVM is aware of that limit. In K8s deployment, we can set the resource limits for containers and inject these limits to 
+some environment variables by downward API. When the container starts to run, it can pick up the environment variables and
+set JVM options accordingly.
+
+We also want to add liveness check and readiness check in the production environment. Spring Boot application
+has end points such as `/actuator/health` for this. We can customize the k8s deployment resource to talk to Spring Boot end point.
+
+Download the patch `patch.yaml`. It contains the memory limits setup and livenessProbes.
+<!-- @downloadPatch @test -->
```
Comment (droot) [inline cmd/kinflate/demo/springboot.md]: was wondering if separating out these patches is better idea. Not a deal breaker, but something to think.

Having each patch do one thing is a good practice to communicate. That way people can turn on/off individual patches (mix-ins kinda behavior).

Comment (Liujingfang1) [reply cmd/kinflate/demo/springboot.md]: It is better. I was thinking the same when I wrote the demo. The reason they were put together is that the file is downloaded from spring boot instance repo, where it is as a single patch file. 
I will separate the patches in the repo and update here.

**Comment by Liujingfang1** [comment]
Update the demo according to Sunil's comments.

File: cmd/kinflate/demos/springboot.md [hunk 9]
```diff
@@ -0,0 +1,350 @@
...
+```
+mv $DEMO_HOME/Kube-manifest.yaml $DEMO_HOME/tmp.yaml
+sed '/patches:$/{N;s/- patch.yaml/- patch.yaml\n- memorylimit_patch.yaml\n- healthcheck_patch.yaml/}' $DEMO_HOME/tmp.yaml >& $DEMO_HOME/Kube-manifest.yaml
+```
+`Kube-manifest.yaml` should have patches field:
+> ```
+> patches
+> - patch.yaml
+> - memorylimit_patch.yaml
+> - healthcheck_patch.yaml
```
Comment (droot) [inline cmd/kinflate/demos/springboot.md]: Nice :)

**Pull request merged**